infra/src/main/resources/db/migration/prod/V77__add_layout_segment_version_statistics.sql

```diff
@@ -0,0 +1,3 @@
+create statistics layout.segment_version_alignment_stats on alignment_id, alignment_version from layout.segment_version;
+alter statistics layout.segment_version_alignment_stats set statistics 1000;
```

Comment:
En ole aiemmin PG:n statistiikkoja joutunut itse käsin tunkkaamaan. Mitäs tämä 1000 käytännössä tarkoittaa, mikä tämän aiempi arvo oli? Ja tarviiko tätä tunkata aina jos/kun esim. `segment_version`-taulun rakenne muuttuu (eli käytännössä kannattaisiko tämän olla repeatable-migraatio?)

Reply:
Arvon nimi on "statistics target", oletusarvo on 100, vaihteluväli 1-10000. En ole ihan varma, tarkoittaako tuon arvon muuttaminen useampaakin asiaa, mutta ainakin siihen se vaikuttaa, kuinka monen rivin näytteen analyze ottaa tilastojen laskemista varten, ja most_common_values-lista on myös juuri statistics targetin pituinen. Käytännössä juuri tässä tapauksessa oli niin, että:

- alkuperäinen taskin esimerkkihaku tämän "create statistics"in kanssa mutta statistics targetin oletusarvolla oli hidas
- ... mutta isommalla arvolla nopea
- ... mutta isommallakaan arvolla juuri esimerkin alignment_id:tä ja alignment_versionia ei ollut tämän statistiikan most_common_values-listassa, eli kyse ei ollut siitä, että tuurilla postgres olisi tiennyt suoraan juuri tämän alignmentin version segmenttien määrän

Tuosta voisi ottaa vielä mielipiteitä vastaan, kannattaisiko tämän nyt olla repeatable. Periaatteessahan statistiikka on postgressissa olio tietokannassa siinä missä muutkin.

Reply:
Kuulostaisi minun korvaan hyvältä että tämmöiset olisi repeatablena, jotta niitä voidaan haluttaessa räplätä paremmiksi. Oikeastaan melkein kaiken kannattaa olla repeatable jos se vain voi olla, eli jos sillä ei ole järestys-riippuvuutta (toki aina voi olettaa että versioidut on jo ajettu) ja sen voi kirjoittaa idempotentitiksi niin että muutokset toimii. Tämän nähdäkseni voi, koska statistiikka voidaan aina vain hukata ja luoda uusiksi.